### PR TITLE
Add support new navigation for maps

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
Add support new navigation for maps

### Issues Resolved
Maps for https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7029

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin.tsx [hunk 1]
```diff
@@ -100,6 +102,26 @@ export class CustomImportMapPlugin
...
+      order: 200,
+    }]);
+
+    core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.analytics, [{
+      id: MAPS_APP_ID,
+      category: DEFAULT_APP_CATEGORIES.dashboardAndReport,
+      order: 200,
+      // Pending change ready from OSD
+      // showInAllNavGroup: true,
+    }]);
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:117]: Hi Junqiu, confirmed that analytics won't have Maps, could you please add Maps to security analytics?

File: public/plugin.tsx [hunk 2]
```diff
@@ -100,6 +102,26 @@ export class CustomImportMapPlugin
       },
     });
 
+    core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.observability, [{
+      id: MAPS_APP_ID,
+      category: DEFAULT_APP_CATEGORIES.dashboardAndReport,
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:107]:
```suggestion
      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
```

File: public/plugin.tsx [hunk 3]
```diff
@@ -100,6 +102,26 @@ export class CustomImportMapPlugin
...
+      order: 200,
+    }]);
+
+    core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.analytics, [{
+      id: MAPS_APP_ID,
+      category: DEFAULT_APP_CATEGORIES.dashboardAndReport,
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:113]:
```suggestion
      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
```

File: public/plugin.tsx [hunk 4]
```diff
@@ -100,6 +102,26 @@ export class CustomImportMapPlugin
...
+      order: 200,
+      // Pending change ready from OSD
+      // showInAllNavGroup: true,
+    }]);
+
+    core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.search, [{
+      id: MAPS_APP_ID,
+      category: DEFAULT_APP_CATEGORIES.dashboardAndReport,
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:121]:
```suggestion
      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
```

File: public/plugin.tsx (hunk 1 above)
Comment (junqiu-lei) [reply public/plugin.tsx:117]: Ack, thanks confirm!

Comment (junqiu-lei) [reply public/plugin.tsx]: Then do we still need put 
```
// Pending change ready from OSD
// showInAllNavGroup: true,
```
for the `security analytics` section? @SuZhou-Joe

Comment (junqiu-lei) [reply public/plugin.tsx]: Ack, we can add it latter once confirmed

File: public/plugin.tsx [hunk 5]
```diff
@@ -100,6 +102,24 @@ export class CustomImportMapPlugin
...
+
+    core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS['security-analytics'], [{
+      id: MAPS_APP_ID,
+      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
+      order: 200,
+    }]);
+
+    core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.search, [{
+      id: MAPS_APP_ID,
+      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:119]:
```suggestion
      category: DEFAULT_APP_CATEGORIES.analyzeSearch,
```
Seems the category should be DEFAULT_APP_CATEGORIES.analyzeSearch for search use case.

**Review by SuZhou-Joe — APPROVED** [review]

File: public/plugin.tsx (hunk 5 above)
Comment (junqiu-lei) [reply public/plugin.tsx]: Updated

**Review by SuZhou-Joe — APPROVED** [review]

**Comment by junqiu-lei** [comment]
The test failure is due to CI runner env.

**Review by martin-gaievski — APPROVED** [review]

**Review by heemin32 — APPROVED** [review]

**Pull request merged**